mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -546,6 +736,21 @@ class LogisticRegression @Since("1.2.0") (
             math.log(histogram(1) / histogram(0)))
         }
 
+        if (usingBoundConstrainedOptimization) {
+          // Make sure all initial values locate in the corresponding bound.
+          var i = 0
+          while (i < numCoeffs) {
+            val initialCoefWithInterceptArray =
+              initialCoefWithInterceptMatrix.asInstanceOf[DenseMatrix].values
```

Comment:
Mutating the states implicitly can be dangerous. Can we use `apply, index, and update` apis in matrix?